AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -112,52 +111,44 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
         guard let lastMeasurementTime = parser.getMeasurementTime(lineString: lastLineInFile) else {
             Log.error("Failed to read last measurement time from file")
             completion(.failure(SDMobileSavingErrors.noLastMeasurementTime))
             return
         }
         
         do {
-            try self.fileLineReader.readLines(of: fileURL, progress: { line in
+            try parser.enumerateMeasurements(url: fileURL, action: { measurements in
                 guard !savingFailed else { return }
-                switch line {
-                case .line(let content):
-                    context.perform {
-                        guard let measurements = self.parser.parseMeasurement(lineString: content) else {
-                            Log.error("Failed to parse line from the file: \(content)")
-                            return
-                        }
-                        
-                        // This happens only with the first line
-                        if sessionData.averaging == nil {
-                            sessionData.averaging = self.averagingService.calculateAveragingWindow(startTime: sessionData.startTime ?? measurements.date, lastMeasurement: lastMeasurementTime)
-                        }
-                        
-                        Log.info("[SD sync] \(i) - \(savedLines): \(measurements.date)") // TO DELETE AFTER TESTING
-                        i += 1
-                        guard sessionData.lastMeasurementTime == nil || measurements.date > sessionData.lastMeasurementTime! else {
-                            return
-                        }
-                        
-                        self.enqueueForSaving(measurements: measurements, buffer: &streamsWithMeasurements, deviceID: deviceID)
-                        
-                        savedLines += 1
-                        guard savedLines == bufferThreshold else { return }
-                        
-                        do {
-                            Log.info("Threshold exceeded, saving data")
-                            try self.saveData(streamsWithMeasurements, session: &sessionData)
-                            streamsWithMeasurements = [:]
-                            savedLines = 0
-                        } catch {
-                            Log.error("Saving measurements failed")
-                            savingFailed = true
-                            return
-                        }
+                
+                context.perform {
+                    // This happens only with the first line
+                    if sessionData.averaging == nil {
+                        sessionData.averaging = self.averagingService.calculateAveragingWindow(startTime: sessionData.startTime ?? measurements.date, lastMeasurement: lastMeasurementTime)
+                    }
+                    
+                    Log.info("[SD sync] \(i) - \(savedLines): \(measurements.date)") // TO DELETE AFTER TESTING
```

Comment:
Please bear in mind before merging to develop \ or after testing with Michael - when omit 😄

Reply:
oh thanks, removing it now